project/src/com/google/daggerquery/executor/models/Query.java

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.daggerquery.executor.models;
+
+/**
+ * A class which represents one query.
+ *
+ * <p>Each query has a {@code type} which itself an instance of {@link QueryType}.
+ * There cannot be two queries with the same type.
```

Comment:
I'm confused by this sentence :confused: 

It seems to suggest I cannot ask `deps Cat` followed by `deps Dogs`. Maybe remove?

Reply:
I mean, `QueryType` can be the id of each query, but yes, I wrote it not clear and this sentence is quite ambiguous. Anyway, I am saying about this fact in javadoc for `QueryType` class. Removed ✅